crates/router/src/routes/user.rs

```diff
@@ -39,15 +39,19 @@ pub async fn user_signup_with_merchant_id(
     state: web::Data<AppState>,
     http_req: HttpRequest,
     json_payload: web::Json<user_api::SignUpWithMerchantIdRequest>,
+    query: web::Query<user_api::AuthIdQueryParam>,
 ) -> HttpResponse {
     let flow = Flow::UserSignUpWithMerchantId;
     let req_payload = json_payload.into_inner();
+    let auth_id = query.into_inner().auth_id;
     Box::pin(api::server_wrap(
         flow.clone(),
         state,
         &http_req,
         req_payload.clone(),
-        |state, _, req_body, _| user_core::signup_with_merchant_id(state, req_body),
+        |state, _, req_body, _| {
+            user_core::signup_with_merchant_id(state, req_body, auth_id.clone())
```

Comment:
`clone()` not required.

Reply:
Will pickup in upcoming PRs.

Reply:
Unable to do this, as the `ApiMetricEvent` should be implemented for payload. That means, if I use a tuple like `(req, auth_id)`, then I have to implement `ApiMetricEvent` for `(ReqType, String)`